[WebKitLegacy] [MiniBrowser] Web Inspector window does not close when inspected page window is closed.

basuke opened the pull request:
#### e5579bd106dfccd15c1c1cc25ac86150b8354207
<pre>
[WebKitLegacy] [MiniBrowser] Web Inspector window does not close when inspected page window is closed.
<a href="https://bugs.webkit.org/show_bug.cgi?id=303576">https://bugs.webkit.org/show_bug.cgi?id=303576</a>
<a href="https://rdar.apple.com/165858296">rdar://165858296</a>

Reviewed by Alexey Proskuryakov.

`WK1BrowserWindowController::windowWillClose:` does not call `[_webView close]`, which prevents:
1. WebView cleanup and deallocation
2. Inspector notification via `inspectedWebViewClosed`
3. Proper Inspector window closure

Without calling `[_webView close]`, both the WebView and WK1BrowserWindowController leak memory
and the Inspector close flow is never triggered.

Call `[_webView close]` in `WK1BrowserWindowController::windowWillClose:` before notifying the
app delegate. This ensures proper cleanup of the WebView and triggers the Inspector close flow:

WebView::close → WebView::_close → inspectedWebViewClosed → WebInspector::close →
WebInspectorFrontend::close → WebInspectorFrontendClient::closeWindow → Inspector window closes

* Tools/MiniBrowser/mac/WK1BrowserWindowController.m:
(-[WK1BrowserWindowController windowWillClose:]):

Canonical link: <a href="https://commits.webkit.org/304008@main">https://commits.webkit.org/304008@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f976017dd255391d33ee71fbbd792a7f34471815

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134085 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6596 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45291 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141665 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86147 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3d07b1e2-e806-4a52-bf3c-27253998fa44) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135955 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7130 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6460 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102556 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/69841 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137032 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4990 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120198 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83351 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4869 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2496 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/1480 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114040 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38321 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144311 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6266 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38900 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110921 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6348 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5293 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111148 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28230 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4729 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116459 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60036 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6318 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34693 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6164 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6409 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6272 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Generally, I think that we should be moving towards removing support for Web Inspector in legacy WebKit, but this seems fine to land.

basuke committed:
[WebKitLegacy] [MiniBrowser] Web Inspector window does not close when inspected page window is closed.
https://bugs.webkit.org/show_bug.cgi?id=303576
rdar://165858296

Reviewed by Alexey Proskuryakov.

`WK1BrowserWindowController::windowWillClose:` does not call `[_webView close]`, which prevents:
1. WebView cleanup and deallocation
2. Inspector notification via `inspectedWebViewClosed`
3. Proper Inspector window closure

Without calling `[_webView close]`, both the WebView and WK1BrowserWindowController leak memory
and the Inspector close flow is never triggered.

Call `[_webView close]` in `WK1BrowserWindowController::windowWillClose:` before notifying the
app delegate. This ensures proper cleanup of the WebView and triggers the Inspector close flow:

WebView::close → WebView::_close → inspectedWebViewClosed → WebInspector::close →
WebInspectorFrontend::close → WebInspectorFrontendClient::closeWindow → Inspector window closes

* Tools/MiniBrowser/mac/WK1BrowserWindowController.m:
(-[WK1BrowserWindowController windowWillClose:]):

Canonical link: https://commits.webkit.org/304008@main
